sympy/calculus/tests/test_util.py

```diff
@@ -21,6 +21,7 @@ def test_function_range():
     assert function_range(exp(x), x, Interval(-1, 1)) == Interval(exp(-1), exp(1))
     assert function_range(log(x) - x, x, S.Reals) == Interval(-oo, -1)
     assert function_range(sqrt(3*x - 1), x, Interval(0, 2)) == Interval(0, sqrt(5))
+    raises(NotImplementedError, lambda : function_range((a*x + b)*(exp(x) - 3), x, S.Reals))
```

Comment:
see note on #13121 -- rather than having to patch this test some day, can you just use something that we know doesn't have a solution, perhaps `f = sin(x)*exp(x) - 1`?

Reply:
I found another issue #13273, while using `f = sin(x)*exp(x) - 1`

Reply:
Also the differentiation of `f` is solvable by `solveset`:

```ipython
In [11]: f = sin(x)*exp(x) - 1
In [14]: solveset(f.diff(), x, S.Reals)
Out[14]: Union(ImageSet(Lambda(_n, 2*_n*pi + 3*pi/4), S.Integers), ImageSet(Lambda(_n, 2*_n*pi + 7*pi/4), S.Integers))
```

Reply:
Forgot about the derivative...ok, how about `f = exp(x)*(sin(x)-cos(x))/2 - x` (whose derivative is the function I suggested).

Reply:
Yeah, that's better. Fixed now.